ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/Extractor.kt

```diff
@@ -2,9 +2,17 @@ package org.gradle.devprod.collector.enterprise.export.extractor
 
 import org.gradle.devprod.collector.enterprise.export.model.BuildEvent
 
-abstract class Extractor<T>(val eventType: String) {
-    abstract fun extract(events: Iterable<BuildEvent>): T
+abstract class Extractor<T>(val eventTypes: List<String>) {
+    open fun extract(events: Iterable<BuildEvent>): T {
+        throw NotImplementedError()
+    }
 
-    fun extractFrom(events: Map<String?, List<BuildEvent>>): T =
-        extract(events.getOrDefault(eventType, listOf()))
+    open fun extractFrom(events: Map<String?, List<BuildEvent>>): T {
```

Comment:
```suggestion
    abstract fun extractFrom(events: Map<String?, List<BuildEvent>>): T
```

Why not make this method abstract?